### ISSUE TITLE
`vacuum` doesn't detect present `info.contact` and `info.license` objects

### ISSUE THREAD

**Issue by joe-mann** [daveshanley/vacuum#228]
Using the sample OpenAPI spec below, with valid `info.contact` and `info.license` objects, vacuum (v0.0.49) reports the following warnings:

```json
{
    "message": "Info section is missing contact details: `contact` must be set",
    "range": {
        "start": {
            "line": 3,
            "character": 3
        },
        "end": {
            "line": 11,
            "character": 5
        }
    },
    "path": "$.info",
    "ruleId": "info-contact",
    "ruleSeverity": "warn"
},
{
    "message": "Info section should contain a license: `license` must be set",
    "range": {
        "start": {
            "line": 3,
            "character": 3
        },
        "end": {
            "line": 11,
            "character": 5
        }
    },
    "path": "$.info",
    "ruleId": "info-license",
    "ruleSeverity": "info"
}
```

Sample spec:

```yaml
openapi: "3.0.0"
info:
  version: 1.0.0
  title: Swagger Petstore
  description: The Petstore REST API.
  contact:
    name: API team
    url: https://smartbear.com/contact-us/
    email: info@SmartBear.com
  license:
    name: MIT
    url: https://opensource.org/licenses/MIT
servers:
  - url: http://petstore.swagger.io/v1
paths:
  /pets:
    get:
      summary: List all pets
      operationId: listPets
      tags:
        - pets
      parameters:
        - name: limit
          in: query
          description: How many items to return at one time (max 100)
          required: false
          schema:
            type: integer
            format: int32
      responses:
        200:
          description: An paged array of pets
          headers:
            x-next:
              description: A link to the next page of responses
              schema:
                type: string
          content:
            application/json:    
              schema:
                $ref: "#/components/schemas/Pets"
        default:
          description: unexpected error
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/Error"
    post:
      summary: Create a pet
      operationId: createPets
      tags:
        - pets
      responses:
        201:
          description: Null response
        default:
          description: unexpected error
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/Error"
  /pets/{petId}:
    get:
      summary: Info for a specific pet
      operationId: showPetById
      tags:
        - pets
      parameters:
        - name: petId
          in: path
          required: true
          description: The id of the pet to retrieve
          schema:
            type: string
      responses:
        200:
          description: Expected response to a valid request
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/Pets"
        default:
          description: unexpected error
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/Error"
components:
  schemas:
    Pet:
      required:
        - id
        - name
      properties:
        id:
          type: integer
          format: int64
        name:
          type: string
        tag:
          type: string
    Pets:
      type: array
      items:
        $ref: "#/components/schemas/Pet"
    Error:
      required:
        - code
        - message
      properties:
        code:
          type: integer
          format: int32
        message:
          type: string
```

**Comment by daveshanley** [comment]
This seems odd; looking into it for you.

**Comment by daveshanley** [comment]
**Unfortunately, I can't reproduce this issue.** 

The rules are not triggered when I run the following command:

`vacuum lint spec.yaml -d`
![image](https://user-images.githubusercontent.com/187345/210584061-677c1822-08e0-4324-a91c-e2ce875cc1a3.png)

To verify those rules are not being generated by the report, run this command:
 
`vacuum report -o spec.yaml | grep "info-contact"`

No results.

**Comment by joe-mann** [comment]
Hi @daveshanley, apologises for the long delay.

The problem only appears to happen when using a `--ruleset`. Using exact spec in my original comment, and having `all-ruleset.yaml` containing the following:

```yaml
extends: [[spectral:oas, all]]
```

I run `vacuum lint spec.yaml --ruleset all-ruleset.yaml` and the warnings appear as originally described. If I remove the `--ruleset` flag, `vacuum` reports correctly your own findings.

**Comment by CB-lferreira** [comment]
Hi @daveshanley, 
I also have the exact same problem, using vacuum (v0.1.4).
It really only happens when using a `--ruleset` that is  not the recommended one. Any other will not detect `info.contact` objects.

**Comment by daveshanley** [comment]
This should be fixed in `v0.2.2` Would someone be able to verify? @CB-lferreira @joe-mann

**Comment by daveshanley** [comment]
I am closing for now as I cannot reproduce this anymore.
